website/pages/docs/commands/operator/snapshot-agent.mdx

```diff
@@ -0,0 +1,149 @@
+description: |
+  Periodically saves snapshots of Nomad server state
+---
+
+# Command: operator snapshot agent
+
+<EnterpriseAlert />
+
+The snapshot agent takes snapshots of the state of the nomad servers and
+saves them locally, or pushes them to an optional remote storage service.
+
+The agent can be run as a long-running daemon process or in a one-shot mode
+from a batch job. As a long-running daemon, the agent will perform a leader
+election so multiple processes can be run in a highly available fashion with
+automatic failover. In daemon mode, the agent will also register itself with
+Nomad as a service, along with health checks that show the agent is alive
+and able to take snapshots.
+
+If ACLs are enabled, a management token must be supplied in order to perform
+snapshot operations.
+
+The Config file has the following format (shown populated with default values):
+
+```hcl
+nomad {
+  http_addr       = "127.0.0.1:8500"
+  token           = ""
+  region          = ""
+  ca_file         = ""
+  ca_path         = ""
+  cert_file       = ""
+  key_file        = ""
+  tls_server_name = ""
+}
+
+
+snapshot {
+  interval         = "1h"
+  retain           = 30
+  stale            = false
+  service          = "nomad-snapshot"
+  deregister_after = "72h"
+  lock_key         = "nomad-snapshot/lock"
+  max_failures     = 3
+  name_prefix      = "nomad"
+}
+
+log {
+  level           = "INFO"
+  enable_syslog   = false
+  syslog_facility = "LOCAL0"
+}
+
+consul {
+  enabled         = true
+  http_addr       = "127.0.0.1:8500"
+  token           = ""
+  datacenter      = ""
+  ca_file         = ""
+  ca_path         = ""
+  cert_file       = ""
+  key_file        = ""
+  tls_server_name = ""
+}
+
+# one storage block is required
+
+local_storage {
+  path = "."
+}
+
+aws_storage {
+  access_key_id     = ""
+  secret_access_key = ""
+  s3_region         = ""
+  s3_endpoint       = ""
+  s3_bucket         = ""
+  s3_key_prefix     = "nomad-snapshot"
+}
+
+azure_blob_storage {
+  account_name   = ""
+  account_key    = ""
+  container_name = ""
+}
+
+google_storage {
+  bucket = ""
+}
+```
+
+
+## Usage
+
+```plaintext
+nomad operator snapshot agent [options] <config_file>
+```
+
+## General Options
+
+@include 'general_options.mdx'
+
+## Snapshot agent Options
+
+### Snapshot Options
+- `-interval`: Interval at which to perform snapshots as a time with a unit suffix, which can be "s", "m", "h" for seconds, minutes, or hours. If 0 is provided, the agent will take a single snapshot and then exit, which is useful for running snapshots via batch jobs. Defaults to "1h".
+- `-lock-key`: A prefix in Consul's key-value store used to coordinate between different instances of the snapshot agent in order to only have one active instance at a time. For highly available operation of the snapshot agent, simply run multiple instances. All instances must be configured with the same lock key in order to properly coordinate. Defaults to "nomad-snapshot/lock".
```

Comment:
```suggestion
- `-lock-key`: A prefix in Nomad's key-value store used to coordinate between different instances of the snapshot agent in order to only have one active instance at a time. For highly available operation of the snapshot agent, simply run multiple instances. All instances must be configured with the same lock key in order to properly coordinate. Defaults to "nomad-snapshot/lock".
```

Reply:
Actually - upon reading this again - yes, this is meant to be Consul not Nomad :( will revert this change